File: frontend/src/api/sessions.ts
// 会话 API（单一会话模式 + 归档）
import { apiFetch, getBaseUrl } from './index'

export interface SessionInfo {
  id: string
  title: string
  created_at: number
  updated_at: number
  count: number
}

export interface Message {
  role: 'user' | 'bot'
  content: string
  image?: string | null
  ts: number
}

export interface ArchiveInfo {
  id: string
  title: string
  created_at: number
  message_count: number
}

export interface ArchiveDetail extends ArchiveInfo {
  messages: Message[]
}

/** 归档搜索结果摘要（列表展示用，不含完整消息）。 */
export interface ArchiveSearchResult {
  id: string
  title: string
  created_at: number
  message_count: number
  preview: string
}

/** 彻底重置（失忆重开）的返回统计。 */
export interface ResetStats {
  userdb_tables: number
  vector: number
  session_msgs: number
}

// 单一会话固定 id（与后端 CURRENT_SESSION_ID 对齐）
export const CURRENT_SESSION_ID = 'current'

export async function getMessages(id: string): Promise<Message[]> {
  const r = await apiFetch(`/api/sessions/${id}`)
  if (!r.ok) return []
  return r.json()
}

// ---- 归档 ----

/** 结束并归档当前会话：后端打包当前消息存入归档库，清空当前会话。 */
export async function archiveCurrent(): Promise<ArchiveInfo | null> {
  const r = await apiFetch('/api/sessions/archive', { method: 'POST' })
  const d = await r.json()
  if (!r.ok || !d.archived) return null
  return d.archive ?? null
}

/** 彻底重置（失忆重开）：清空菟菚积累的记忆/好感/昵称/向量 + 当前会话气泡。 */
export async function resetUser(): Promise<ResetStats> {
  const r = await apiFetch('/api/user/reset', { method: 'POST' })
  const d = await r.json()
  if (!r.ok || !d.ok) throw new Error(d.error || '重置失败')
  return d.reset ?? { userdb_tables: 0, vector: 0, session_msgs: 0 }
}

/** 归档列表（只读回看）。 */
export async function listArchives(): Promise<ArchiveInfo[]> {
  const r = await apiFetch('/api/sessions/archives')
  if (!r.ok) return []
  const d = await r.json()
  return d.archives ?? []
}

/** 单条归档详情（含完整消息）。 */
export async function getArchive(id: string): Promise<ArchiveDetail | null> {
  const r = await apiFetch(`/api/sessions/archives/${id}`)
  if (!r.ok) return null
  const d = await r.json()
  return d.archive ?? null
}

/** 按关键词搜索归档（标题 + 内容），后端返回摘要列表（标题/条数/命中预览）。 */
export async function searchArchives(q: string): Promise<ArchiveSearchResult[]> {
  const r = await apiFetch(`/api/sessions/archives/search?q=${encodeURIComponent(q)}`)
  if (!r.ok) return []
  const d = await r.json()
  return d.results ?? []
}

// 菟菚主动消息：后端已生成、待投递队列里的一条（取走后即清空）
export async function getInitiative(sessionId: string): Promise<string | null> {
  const r = await apiFetch(`/api/initiative?session_id=${encodeURIComponent(sessionId)}`)
  const d = await r.json()
  return d.initiative ?? null
}

// 菟菚主动消息 SSE 长连接：服务端后台生成主动消息时秒级推送（替代 30s 轮询）
export function openInitiativeStream(
  sessionId: string,
  onMessage: (text: string) => void,
  onError?: () => void,
): EventSource {
  const url = `${getBaseUrl()}/api/initiative/stream?session_id=${encodeURIComponent(sessionId)}`
  const es = new EventSource(url)
  es.addEventListener('initiative', (e: MessageEvent) => {
    try {
      const data = JSON.parse(e.data)
      if (data && data.text) onMessage(data.text)
    } catch {
      // 解析失败忽略
    }
  })
  if (onError) es.onerror = onError
  return es
}

File: deploy/TZtuzhanAssistant-Deploy-Full-v2.0.0-大杯版/backend/api/sessions.py
# -*- coding: utf-8 -*-
"""会话接口 + 归档接口（单一会话模式）。

单一会话模式下只有一个固定会话（id='current'），因此不再提供
新建/删除/重命名/列表/搜索等多会话 CRUD，仅保留：
- 归档（结束并归档当前会话 / 归档列表 / 归档详情）
- 读取当前会话消息（GET /{session_id}，session_id 恒为 'current'）
- 导出当前会话（markdown / json）
"""
from __future__ import annotations

import datetime as _dt

from fastapi import APIRouter
from fastapi.responses import JSONResponse, Response

from ..session.store import (
    archive_current,
    get_archive,
    get_messages,
    list_archives,
    search_archives,
)

router = APIRouter(prefix="/api/sessions", tags=["sessions"])


# ---- 归档（单一会话模式下新增）----

@router.post("/archive")
async def api_sessions_archive():
    """结束并归档当前会话：打包当前所有消息存入 archives，清空当前会话。"""
    result = await archive_current()
    if result is None:
        return {"ok": True, "archived": False, "message": "当前会话没有可归档的内容"}
    return {"ok": True, "archived": True, "archive": result}


@router.get("/archives")
async def api_archives_list():
    """归档列表（只读回看）。"""
    return {"ok": True, "archives": await list_archives()}


@router.get("/archives/search")
async def api_archives_search(q: str = ""):
    """按关键词搜索归档（标题 + 内容），返回摘要列表（标题/条数/命中预览）。

    只返回列表展示所需字段，不含完整消息；点进详情再走 /archives/{id} 拉完整内容。
    注意：必须声明在 /archives/{archive_id} 之前，否则会被动态路由吞掉。
    """
    results = await search_archives(q)
    return {"ok": True, "results": results}


@router.get("/archives/{archive_id}")
async def api_archives_get(archive_id: str):
    """单条归档详情（含完整消息）。"""
    data = await get_archive(archive_id)
    if data is None:
        return JSONResponse({"ok": False, "error": "归档不存在"}, status_code=404)
    return {"ok": True, "archive": data}


# ---- 单一会话读取 / 导出 ----

@router.get("/{session_id}")
async def api_sessions_get(session_id: str):
    msgs = await get_messages(session_id)
    if msgs is None:
        return JSONResponse({"ok": False, "error": "会话不存在"}, status_code=404)
    return msgs


@router.get("/{session_id}/export")
async def api_sessions_export(session_id: str, fmt: str = "md"):
    """导出会话：fmt=md（Markdown）/ json（原始消息）。"""
    msgs = await get_messages(session_id)
    if msgs is None:
        return JSONResponse({"ok": False, "error": "会话不存在"}, status_code=404)

    title = "新会话"
    if fmt == "json":
        return {
            "ok": True,
            "session_id": session_id,
            "title": title,
            "messages": msgs,
        }
    # Markdown 导出
    lines = [f"# {title}", "", "> 导出自 菟菚 桌面助手", ""]
    for m in msgs:
        who = "你" if m["role"] == "user" else "菟菚"
        content = (m.get("content") or "").strip()
        img = m.get("image") or ""
        if not content and img:
            content = "（图片）"
        ts = ""
        if m.get("ts"):
            ts = _dt.datetime.fromtimestamp(m["ts"]).strftime("%Y-%m-%d %H:%M")
        lines.append(f"**{who}**" + (f"（{ts}）" if ts else "") + f": {content}")
        if img:
            lines.append(f"![]({img})")
        lines.append("")
    return Response(
        "\n".join(lines),
        media_type="text/markdown; charset=utf-8",
        headers={"Content-Disposition": f'attachment; filename="session-{session_id}.md"'},
    )
